src/connection.cpp

```diff
@@ -373,25 +373,31 @@ namespace oxen::quic
     }
 
     std::shared_ptr<Stream> Connection::construct_stream(
-            const std::function<std::shared_ptr<Stream>(Connection& c, Endpoint& e)>& make_stream,
+            const std::function<std::shared_ptr<Stream>(Connection& c, Endpoint& e)>& default_stream,
             std::optional<int64_t> stream_id)
     {
         std::shared_ptr<Stream> stream;
         if (context->stream_construct_cb)
             stream = context->stream_construct_cb(*this, _endpoint, stream_id);
+        if (!stream && default_stream)
+            stream = default_stream(*this, _endpoint);
         if (!stream)
-            stream = make_stream(*this, _endpoint);
+            stream = _endpoint.make_shared<Stream>(*this, _endpoint, context->stream_data_cb, context->stream_close_cb);
 
         return stream;
     }
 
-    std::shared_ptr<Stream> Connection::queue_stream_impl(
+    std::shared_ptr<Stream> Connection::queue_incoming_stream_impl(
             std::function<std::shared_ptr<Stream>(Connection& c, Endpoint& e)> make_stream)
     {
         return _endpoint.call_get([this, &make_stream]() {
-            auto stream = construct_stream(make_stream);
+            std::shared_ptr<Stream> stream;
+            if (make_stream)
+                stream = make_stream(*this, _endpoint);
```

Comment:
Why do you only pass the optional ID to the make_stream in `Connection::stream_opened`? This completely eliminates the use case of queueing streams to be opened in the client's on_conn_open callback, for example